test/distributed/_sharded_tensor/test_sharded_tensor.py

```diff
@@ -207,21 +203,27 @@ def test_rand(self):
 
     @sandcastle_skip_if(torch.cuda.device_count() < 1, 'CUDA GPU is needed')
     def test_full(self):
-        fill_value = 23
-        expected_dtype = torch.int32
-        tensor_init_params = TensorInitParams(
-            create_op=CreateOp.FULL,
-            fill_value=fill_value,
-            dtype=expected_dtype,
+        fill_value = 23.5
+        tensor_properties = TensorProperties(
+            # tensor's dtype can be inferred from fill_value
+            dtype=None,
             layout=torch.strided,
             requires_grad=False,
             pin_memory=False,
-            memory_format=torch.contiguous_format, )
+            memory_format=torch.contiguous_format,
+        )
+        tensor_init_params = TensorInitParams(
+            create_op=CreateOp.FULL,
+            fill_value=fill_value,
+            tensor_properties=tensor_properties, )
         local_device = torch.device('cuda:0')
         h, w = 5, 10
         local_tensor = _create_tensor_from_params(
             h, w, local_device=local_device, tensor_init_params=tensor_init_params)
-        expected_tensor = torch.full((h, w), fill_value=fill_value, device=local_device, dtype=expected_dtype)
+
+        print(f"=== bowangbj === \n local_tensor: {local_tensor}")
```

Comment:
nit: can remove this?

Reply:
for some reason, seems you might look at the pre-lint version :) 
removed for real